### PR TITLE
Exercises section 1 to 3 - corrects some translations

### PR DESCRIPTION
**Pull request by NahuelFerreroG**
# 🤘 Courses PRs

## Type

[x] Translation
[ ] Enhancement
[ ] New Article
[x] Exercises

Link a ClicklUp: https://app.clickup.com/t/85zrp41r3

### CODE REVIEW

File: dart/2.Dart_basics/Excercises/2_exercise.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Exercise 2
+
+Declare two integer variables called x and y and initialize them with any values. Then, use asserts to make sure that x is greater than y, and print the message "Assertion passed" if the assertion is true, or "Assertion failed" if the assertion is false.
```
Comment (maurodibert) [inline dart/2.Dart_basics/Excercises/2_exercise.md:3]: do you think would be correct to use  `x` and `y` to reference them? Meaning code?

File: dart/2.Dart_basics/Excercises/3_solution.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Solution 3
+
+```dart
+void main() {
+  int a = 3;
+  int b = 7;
+  String message = (a%2==0 && b%2==0? 'Both a and b are even':(a%2==0 || b%2==0? 'Either a or b is odd': 'Neither a nor b is even')); // note that here we use two ternary operators one insede the other.
```
Comment (maurodibert) [inline dart/2.Dart_basics/Excercises/3_solution.md:7]:
```suggestion
  String message = (a%2==0 && b%2==0? 'Both a and b are even':(a%2==0 || b%2==0? 'Either a or b is odd': 'Neither a nor b is even')); // note that here we use two ternary operators one inside the other.
```

File: dart/2.Dart_basics/Excercises/4_solution.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  newMessage = newMessage.replaceAll('I', '*');
+  newMessage = newMessage.replaceAll('O', '*');
+  newMessage = newMessage.replaceAll('U', '*');
+  print(newMessage);
+}
+```
+
+Although this code solves the exercise, we are not complying with the DRY rule (we'll talk about it in section 4).
+
+The following code does the same but in a much more efficient way. Here we can see that a new concept is introduced: `RegExp()` to solve it. Of course you already know what to do: go read the documentation about it!
```
Comment (maurodibert) [inline dart/2.Dart_basics/Excercises/4_solution.md:22]: I think it's super cool to do this kind of exercise but we should create an article to explain how to use RegEx!

File: dart/2.Dart_basics/Excercises/4_solution.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Solution 4
```
Comment (maurodibert) [inline dart/2.Dart_basics/Excercises/4_solution.md:1]: It's cool but this exercise give me the idea of making an encryption machine as exercise: 

1. you replace one vowel for a symbol.
2. you could enter a text and will be 'encrypted' or viceversa.

What do you think?

File: dart/2.Dart_basics/Excercises/5_solution.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Solution 5
+
+```dart
+void main() {
+  String text = ('This is a newline \nand this is a tab\t and this is a backlash \\');
```
Comment (maurodibert) [inline dart/2.Dart_basics/Excercises/5_solution.md:5]: Great but update the original course (spanish and english) to introduce the ones that are not mentioned ;-)

File: dart/2.Dart_basics/Excercises/8_solution.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
+# Solution 8
```
Comment (maurodibert) [inline dart/2.Dart_basics/Excercises/8_solution.md:1]: Reaaaally cool! 💪

**Review by maurodibert — COMMENTED** [review]
Super good and inspired work Nahu! ❤️ Please continue with this quality!

Some Notes:
- when declaring variables in exercises maybe use `variable`
- If there is something that you introduce that is new to the user like RegEx or new ways of escaping, create an article (first case) or update the ones we have (second case)

File: dart/2.Dart_basics/Excercises/2_exercise.md [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+## Enable asserts
+
+It must be noted that if you want to use assert then you have to enable it while execution as it can only be used in the development mode and not in production mode. If it is not enabled then it will be simply be ignored while execution.
```
[the remark this answers is not in the document]
Comment (NahuelFerreroG) [reply dart/2.Dart_basics/Excercises/2_exercise.md:7]: done!

File: dart/2.Dart_basics/Excercises/3_solution.md (hunk 2 above)
Comment (NahuelFerreroG) [reply dart/2.Dart_basics/Excercises/3_solution.md:7]: done

File: dart/2.Dart_basics/Excercises/4_solution.md (hunk 3 above)
Comment (NahuelFerreroG) [reply dart/2.Dart_basics/Excercises/4_solution.md:22]: Ok! I'll work on it, where do you think it should be? I believe it doesn't fit in any section. Maybe create a new section called 'Interesting commands' or something like that where we add specific articles about this kind of commands and the in the exercise put a likn to it.

File: dart/2.Dart_basics/Excercises/5_solution.md (hunk 5 above)
Comment (NahuelFerreroG) [reply dart/2.Dart_basics/Excercises/5_solution.md:5]: The `\`to scape is introduced, the only new thing here is the `\t` to make a tabulation. I could not find others to add -.-. Anyway i can add that one in the proper article

File: dart/2.Dart_basics/Excercises/4_solution.md (hunk 4 above)
Comment (NahuelFerreroG) [reply dart/2.Dart_basics/Excercises/4_solution.md:1]: great! I'll write it :)

File: dart/2.Dart_basics/Excercises/5_solution.md (hunk 5 above)
Comment (maurodibert) [reply dart/2.Dart_basics/Excercises/5_solution.md:5]: Yeah! Thanks!

File: dart/2.Dart_basics/Excercises/4_solution.md (hunk 4 above)
Comment (maurodibert) [reply dart/2.Dart_basics/Excercises/4_solution.md:1]: I love that enthusiasm!

File: dart/2.Dart_basics/Excercises/4_solution.md (hunk 3 above)
Comment (maurodibert) [reply dart/2.Dart_basics/Excercises/4_solution.md:22]: You know that we have or we'll have, a part of our web which is curated learning courses, meaning a list of courses in order to learn. Other place, is isolated articles. In that part I think should go.

Comment (maurodibert) [reply dart/2.Dart_basics/Excercises/4_solution.md:22]: (Feel free to resolve this - I've left it for you to read the comment)

**Review by maurodibert — APPROVED** [review]
Great work Nahu! I've left you a comment just for you to read but the PR is perfect!

**Pull request merged**